plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetViewer.java

```diff
@@ -1642,8 +1646,37 @@ public void redrawData(boolean attributesChanged, boolean rowsChanged)
         //this.updateStatusMessage();
     }
 
+    private final UIJob statusBarLayoutJob = new UIJob("Pending resultset view status bar relayout") {
+        @Override
+        public IStatus runInUIThread(IProgressMonitor monitor) {
+            boolean changed = false;
+            for (ToolBarManager toolbarMan: toolbarList) {
+                ToolBar toolbar = toolbarMan.getControl();
+                boolean wasCollapsed = toolbar.getData() != null;
+                toolbar.setLayoutData(toolbar.getItemCount() > 0 ? null : new RowData(0, 0));
+                boolean nowCollapsed = toolbar.getData() != null;
+                changed |= (wasCollapsed == nowCollapsed);
+            }
+            if (changed) {
+                ResultSetViewer.this.getControl().layout(true, true);
+            }
+            return Status.OK_STATUS;
+        }
+    };
+    
+    private void collapseToolbarIfEmpty(ToolBar toolbar) {
```

Comment:
Please specify nullability annotations for the method's parameters and its return type.